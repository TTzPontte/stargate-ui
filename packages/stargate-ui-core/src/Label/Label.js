import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@pontte/stargate-ui-core';
import { useStyles } from '@pontte/stargate-ui-styles';

const styles = () => {
  const label = {
    display: 'inline-flex',
    alignItems: 'center',
  };

  return { label };
};

const Label = (props) => {
  const { children, ...factoryProps } = props;
  const [{ label: classLabel }] = useStyles(styles);

  return (
    <Typography
      element="label"
      variant="body"
      children={children}
      className={classLabel}
      {...factoryProps}
    />
  );
};

Label.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Label;
