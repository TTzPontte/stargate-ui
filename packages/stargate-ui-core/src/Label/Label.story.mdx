import {
  Meta,
  Canvas,
  Status,
  Link,
} from '@pontte/stargate-ui-playground';
import { Typography, SvgIcon } from '@pontte/stargate-ui-core';

import Label from './Label';

<Meta title="Core/Label" component={Label} />

<Typography element="h2" variant="h1">
  Label
</Typography>

<Status type="experimental" />

<Typography element="h2" variant="h3">
  Basic Usage
</Typography>

<Typography>
  You can use <code>Label</code> with any node element.
</Typography>

<Canvas>
  <Story name="Basic Usage">
    <Label>Simple label</Label>
    <Label>
      <SvgIcon children={<path d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z" fill="currentColor" />} />
      Label with <Link kind="core-svgicon--page">SvgIcon</Link> component
    </Label>
  </Story>
</Canvas>
