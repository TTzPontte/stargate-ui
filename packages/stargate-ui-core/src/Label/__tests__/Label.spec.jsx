import React from 'react';

import Label from '../Label';

describe('Label', () => {
  it('should render properly', () => {
    const { container } = render(
      <Label>
        Lorem ipsum dolor
      </Label>
    );

    expect(container).toMatchSnapshot();
  });
});
