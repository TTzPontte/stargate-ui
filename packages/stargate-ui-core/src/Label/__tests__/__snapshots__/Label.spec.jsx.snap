// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render properly 1`] = `
<div>
  <label
    class="stargate-factory-0-2-3 stargate-factory-d0-0-2-4 stargate-typography-0-2-1 stargate-typography-d0-0-2-2"
  >
    Lorem ipsum dolor
  </label>
</div>
`;
